lua/cmp_luasnip/choice.lua

```diff
@@ -0,0 +1,45 @@
+local cmp = require("cmp")
+
+local source = {}
+
+source.new = function()
+	return setmetatable({}, { __index = source })
+end
+
+source.get_keyword_pattern = function()
+	return "\\%([^[:alnum:][:blank:]]\\|\\w\\+\\)"
+end
+
+function source:is_available()
+	return require('luasnip.session').active_choice_node ~= nil
+end
+
+
+function source:get_debug_name()
+	return "luasnip_choice"
+end
+
+function source:complete(_, callback)
+  local items = {}
+
+  for idx,word in pairs(require('luasnip').get_current_choices()) do
+    table.insert(items, {
+      word = word,
+      label = word,
```

Comment:
If a choice has multiline text, that text is truncated at the linebreak, it seems.
Not sure if that should be changed (maybe replace linebreaks with `\\n`, although it's likely that won't look good if there are many lines)